### PR TITLE
Server: Handle n_keep parameter in the request

### PR DESCRIPTION
**Pull request by jkarthic**

### CODE REVIEW

File: examples/server/utils.hpp [hunk 1]
```diff
@@ -371,6 +371,7 @@ static json oaicompat_completion_params_parse(
     llama_params["repeat_last_n"]     = json_value(body,   "repeat_last_n",     default_sparams.penalty_last_n);
     llama_params["ignore_eos"]        = json_value(body,   "ignore_eos",        false);
     llama_params["tfs_z"]             = json_value(body,   "tfs_z",             default_sparams.tfs_z);
+    llama_params["n_keep"]            = json_value(body,   "n_keep",            0);
```
Comment (phymbert) [inline examples/server/utils.hpp:374]: Hello, thanks but @ggerganov  @ngxson  I worry this is actually [not OAI compatible](https://github.com/openai/openai-openapi/blob/master/openapi.yaml#L5901) ?

**Review by ngxson — APPROVED** [review]
LGTM. It's quite surprise to know that server does not have `--n-keep` argument, maybe we need to add that in the future.

File: examples/server/utils.hpp (hunk 1 above)
Comment (ngxson) [reply examples/server/utils.hpp:374]: we can consider it as an "extension" to OAI, for example `tfs_z` or `mirostat` that we're having, they are not available on OAI.

Comment (ngxson) [reply examples/server/utils.hpp:374]: In fact this code is duplicated to the one inside `launch_slot_with_task`, I planned to refactor all of this to one place, maybe I'll this during weekend.

**Review by phymbert — APPROVED** [review]

**Pull request merged**